apps/condo/domains/user/schema/AuthenticateOrRegisterUserWithTokenService.test.js

```diff
@@ -0,0 +1,929 @@
+                    })
+                    const [result] = await authenticateOrRegisterUserWithTokenByTestClient(anonymousClient, {
+                        token: confirmPhoneAction.token,
+                        userType: STAFF,
+                        userData: {
+                            name: faker.name.fullName(),
+                            password: faker.internet.password(16),
+                        },
+                    })
+                    expect(result.token).not.toHaveLength(0)
+                    expect(result.user.id).toBeDefined()
+                    const user = await UserAdminUtils.getOne(adminClient, { id: result.user.id })
+                    expect(user.id).not.toBe(residentUser.id)
+                    expect(user.type).toBe(STAFF)
+                    expect(user.phone).toBe(phone)
+                })
+
+                test('should sign in a staff with a verified phone token if a staff with such a phone is already registered', async () => {
+                    const [confirmPhoneAction] = await createTestConfirmPhoneAction(adminClient, {
+                        isPhoneVerified: true, phone: phone,
+                    })
+                    const [result] = await authenticateOrRegisterUserWithTokenByTestClient(anonymousClient, {
+                        token: confirmPhoneAction.token,
+                        userType: STAFF,
+                        userData: {
+                            name: faker.name.fullName(),
+                            password: faker.internet.password(16),
+                        },
+                    })
+                    expect(result.token).not.toHaveLength(0)
+                    expect(result.user.id).toBeDefined()
+                    const user = await UserAdminUtils.getOne(adminClient, { id: result.user.id })
+                    expect(user.type).toBe(STAFF)
+                    expect(user.id).not.toBe(residentUser.id)
+                    expect(user.phone).toBe(phone)
+
+                    const anonymousClient2 = await makeClient()
+                    const [confirmPhoneAction2] = await createTestConfirmPhoneAction(adminClient, {
+                        isPhoneVerified: true, phone: phone,
+                    })
+                    const [result2] = await authenticateOrRegisterUserWithTokenByTestClient(anonymousClient2, {
+                        token: confirmPhoneAction2.token,
+                        userType: STAFF,
+                    })
+                    expect(result2.token).not.toHaveLength(0)
+                    expect(result2.user.id).toBeDefined()
+                    const user2 = await UserAdminUtils.getOne(adminClient, { id: result2.user.id })
+                    expect(user).toEqual(user2)
+                })
+            })
+
+            test('should throw an error if the user is not registered and the required user data (phone, email, password) is missing', async () => {
```

Comment:
you can also add same test for resident user

Reply:
A resident has a required data - only a phone. And at the moment there is registration only by phone. Therefore, a case where required data will be missed is impossible